cts/cts-fencing.in

```diff
@@ -252,13 +253,23 @@ class Test(object):
         """ Clean up the host after executing a test """
 
         if self.stonith_process:
-            self.stonith_process.terminate()
-            self.stonith_process.wait()
+            if self.stonith_process.poll() == None:
+                self.stonith_process.terminate()
+                self.stonith_process.wait()
+            else:
+                return_code = {
+                    getattr(signal, _signame): _signame
+                        for _signame in dir(signal)
+                        if _signame.startswith('SIG') and not _signame.startswith("SIG_")
+                }.get(-self.stonith_process.returncode, "RET=%d" % (self.stonith_process.returncode))
+                msg = "FAILURE - '%s' failed. pacemaker-fenced abnormally exited during test (%s)."
+                self.result_txt = msg % (self.name, return_code)
+                self.result_exitcode = CrmExit.ERROR
 
         self.stonith_output = ""
         self.stonith_process = None
 
-        logfile = io.open('/tmp/stonith-regression.log', 'rt')
+        logfile = io.open('/tmp/stonith-regression.log', 'rt', encoding = "ISO-8859-1")
```

Comment:
A comment explaining why the encoding is needed will likely save someone a lot of confusion years from now  :)

I think the signal handling part should be in its own commit